### PR TITLE
Implement experiment delete  for api v2

### PR DESCRIPTION
**Pull request by cosa65**
# Description

# Details
#### URL to issue
https://biomage.atlassian.net/browse/BIOMAGE-1806

https://ui-martinfosco-ui701-api345.scp-staging.biomage.net/
#### Link to staging deployment URL (or set N/A)
N/A (run in api)

#### Links to any PRs or resources related to this PR
https://github.com/hms-dbmi-cellenics/api/pull/345

#### Integration test branch
master

# Merge checklist
Your changes will be ready for merging after all of the steps below have been completed.

### Code updates
Have best practices and ongoing refactors being observed in this PR
- [ ] Migrated any selector / reducer used to the new format.

### Manual/unit testing
- [ ] Tested changes using InfraMock locally **or** no tests required for change, e.g. Kubernetes chart updates.
- [ ] Validated that current unit tests for code work as expected and are sufficient for code coverage **or** no unit tests required for change, e.g. documentation update.
- [ ] Unit tests written **or** no unit tests required for change, e.g. documentation update.

### Integration testing
**You must check the box below** to run integration tests on the latest commit on your PR branch.
Integration tests have to pass before the PR can be merged. Without checking the box, your PR
**will not pass** the required status checks for merging.

- [ ] Started end-to-end tests on the latest commit.

### Documentation updates
- [ ] Relevant Github READMEs updated **or** no GitHub README updates required.
- [ ] Relevant Wiki pages created/updated **or** no Wiki updates required.

### Optional
- [ ] Staging environment is unstaged before merging.
- [ ] Photo of a cute animal attached to this PR.

### CODE REVIEW

File: src/components/data-management/ProjectCard.jsx [hunk 1]
```diff
@@ -58,7 +61,11 @@ const ProjectCard = (props) => {
...
   const updateProjectName = (newName) => {
     dispatch(updateProject(project.uuid, { name: newName.trim() }));
-    if (projects.meta.error === false) {
+
+    if (
+      (config.currentApiVersion === api.V1 && projects.meta.error === false)
+      || config.currentApiVersion === api.V2
+    ) {
       dispatch(updateExperiment(projectExperiment, { name: newName.trim() }));
     }
```
Comment (cosa65) [inline src/components/data-management/ProjectCard.jsx:70]: Small fix: trigger `updateExperiment` if we are using api v2 (to send the api request, updateProject doesn't trigger any v2 requests)

File: src/redux/actions/experiments/reorderSamples.js [hunk 2]
```diff
@@ -37,9 +37,6 @@ const reorderSamples = (
       },
     );
 
-    const json = await response.json();
-    throwIfRequestFailed(response, json, endUserMessages.ERROR_SAVING);
```
Comment (cosa65) [inline src/redux/actions/experiments/reorderSamples.js:41]: Another small fix: when the error handling was refactored this was missing so adapt it to the new error handling

File: src/components/data-management/ProjectCard.jsx (hunk 1 above)
Comment (ivababukova) [reply src/components/data-management/ProjectCard.jsx:70]: Sure, but then why are you triggering `updateExperiment` for api.V1 too?

Comment (cosa65) [reply src/components/data-management/ProjectCard.jsx:70]: Becuase api v1 doesn't use updateExperiment, it sends a request over `updateProject`

Comment (cosa65) [reply src/components/data-management/ProjectCard.jsx:70]: I am triggering `updateExperiment` for v1 too, `if projects.meta.error === false`, (I don't know why that if is there, but it was there before so I just left it)

File: src/components/EditableField.jsx [hunk 3]
```diff
@@ -96,6 +96,7 @@ const EditableField = (props) => {
       return (
         <>
           <Input
+            data-testid='editableFieldInput'
```
Comment (ivababukova) [inline src/components/EditableField.jsx:99]: What is this used for? Is it for the integration tests?

Comment (cosa65) [reply src/components/EditableField.jsx:99]: no, for the unit tests, Im using `getByTestId`  from rtl

**Review by ivababukova — APPROVED** [review]

**Pull request merged**